### DISCUSSION TITLE
Rounded corners for StyleBoxTexture

### DISCUSSION THREAD

**Discussion by Swarkin** [godotengine/godot-proposals#6846] [GUI]
I'm currently trying to make a button look modern using a gradient and rounded corners.

StyleBoxTexture seems to be missing a bunch of options which the StyleBoxFlat has, meaning I can either do a button with rounded corners and a solid color, or a gradient button without rounded corners.
It would be really nice if I could avoid creating .pngs in third party programs just to have round corners on an UI element, and instead do everything inside of Godot with a GradientTexture and some Theme Override magic.

Everything I've tried has led to this;
![image](https://github.com/godotengine/godot-proposals/assets/102416174/4d589d82-d142-4480-8dba-8d83fe72a662)

I've even tried setting `clip_children` to `Clip + Draw` and making a TextureRect with this gradient, which then led to the button's shadows receiving the white highlight, resulting in the same issue as in the picture above.

**Comment by Zireael07** [comment]
Already proposed in https://github.com/godotengine/godot-proposals/issues/3494 and at least one other

**Reply by Swarkin** [reply]
In this discussion OP wants to add a shader property to the StyleBox, however this is not what I was going for.
I just need the Border properties of the `StyleBoxFlat` to be present on a `StyleBoxTexture` as well, no shader stuff.

**Reply by Zireael07** [reply]
There is no "just" here - adding pretty much anything to StyleBox requires a pretty thorough rework.

**Comment by Calinou** [comment]
To me, it sounds like we should add the ability to use a gradient as StyleBoxFlat's background instead. StyleBoxTexture is meant to be used when you have premade textures with an entirely distinct art direction, rather than flat design UI.

- See also https://github.com/godotengine/godot-proposals/issues/6812.

**Reply by Swarkin** [reply]
That would be a viable solution as well.

**Reply by davthedev** [reply]
Simple gradients on StyleBoxFlat: done!
See https://github.com/godotengine/godot/pull/77215
